UGENE-6760 Two SV and adding an annotation object to sequence

rasputinkirill opened the pull request.

atiunov committed:
UGENE-6634: Error loading shared libraries on launching "./ugeneui"

rasputinkirill committed:
UGENE-6760

rasputinkirill committed:
UGENE-6760

DmitriiSukhomlinov committed:
UGENE-6718 GUI test fix

atiunov commented on src/corelibs/U2View/src/ov_sequence/AnnotatedDNAView.cpp:
```diff
@@ -914,6 +914,7 @@ QString AnnotatedDNAView::addObject(GObject *o) {
     } else if (o->getGObjectType() == GObjectTypes::ANNOTATION_TABLE) {
         AnnotationTableObject *ao = qobject_cast<AnnotationTableObject *>(o);
         SAFE_POINT(NULL != ao, "Invalid annotation table!", QString::null);
+        checkAndAddObjectToOtherViews(o);
```
Each view should work only with itself. Other views will add the object by themselves, if it is supposed to do.

atiunov committed:
UTI-238: QSpec adapted to work both with Webkit and Qt WebEngine; GUI testing enabled for Qt newer than 5.6 (#81)

UTI-238: QSpec adapted to work both with Webkit and Qt WebEngine; GUI testing enabled for Qt newer than 5.6

atiunov committed:
Build with webkit fix

rasputinkirill committed:
Revert "UGENE-6760"

This reverts commit 8ea6d7d4992cc2d47d914064e48527bc4e03d9d5.

rasputinkirill committed:
Revert "UGENE-6760"

This reverts commit 8ea6d7d4992cc2d47d914064e48527bc4e03d9d5.

atiunov committed:
UTI-395: some tests failed after adding QWebEngine support to QSpec

atiunov committed:
Merge branch 'UTI-395'

atiunov committed:
UTI-384: two tests were fixed: there were not enough space for full text iin label

atiunov committed:
Merge branch 'UTI-384'

yalgaer committed:
UGENE-6743. Improve Go to Position dialog in the SV

yalgaer committed:
UGENE-6713. Scroll bar in the Name List in the AE differs from other scroll bars

yalgaer committed:
UGENE-6744. Navigation menu in the Alignment Editor

smlnkcv committed:
Corrected  GUITest_regression_scenarios_test_6640_4

atiunov committed:
ugene.pro splitted to to pro and pri files

atiunov committed:
An additional dependency for QSpec

rasputinkirill committed:
UGENE-6760

rasputinkirill committed:
UGENE-6760

rasputinkirill commented:
For now i fix only problem described in issue - adding annotation table to sequence view.
IMHO problem with removing annotation table should be resolved as separate issue, because current task already took too much time.

rasputinkirill committed:
UGENE-6455 GUI test fix

atiunov committed:
UGENE-6634: '-no-pie' flag added to executables

atiunov committed:
UGENE-6634: LD_LIBRARY_PATH setting removed from linux launch script 'ugene'

atiunov committed:
Merge pull request #87 from ugeneunipro/UGENE-6634

UGENE-6634: Error loading shared libraries on launching "./ugeneui"

atiunov committed:
UGENE-6634: '-no-pie' flag added to executables (reverted from commit ff05893d80f0245b4199e152322b5b0e7cde020c)

yalgaer committed:
UGENE-6730. AE: wrong selection when deleting column

yalgaer committed:
UGENE-6751. Crash after clicking on the Consensus label in the AE and aligning with MUSCLE

yalgaer committed:
Improve Go to Position dialog in the SV

yalgaer committed:
Fix translations build

yalgaer committed:
UGENE-6742. Appearance menu in the Alignment Editor

rasputinkirill committed:
GUI test added

rasputinkirill committed:
GUI test added

atiunov commented on src/corelibs/U2View/src/ov_sequence/AnnotatedDNAView.cpp:
```diff
@@ -786,6 +786,30 @@ void AnnotatedDNAView::sl_updatePasteAction() {
+    QList<GObject *> objectsToAdd;
+    foreach (GObject *obj, allObjs) {
+        foreach (const GObjectRelation &rel, obj->findRelatedObjectsByRole(ObjectRole_Sequence)) {
+            if (rel.ref.entityRef == o->getEntityRef() && !currentAnnotations.contains(qobject_cast<AnnotationTableObject*>(obj))) {
```
Use GObjectUtils::findObjectsRelatedToObjectByRole instead of checking of all objects.

atiunov commented on src/corelibs/U2View/src/ov_sequence/AnnotatedDNAView.cpp:
```diff
@@ -786,6 +786,30 @@ void AnnotatedDNAView::sl_updatePasteAction() {
+    foreach(GObject* obj, objectsToAdd) {
+        QString error = addObject(obj);
+        if (!error.isEmpty()) {
+            QMessageBox::critical(NULL, tr("Error!"), error);
```
If you have several views, then you'll receive several message boxes. I offer just to write the message to log without additional notification.

atiunov commented on src/corelibs/U2View/src/ov_sequence/AnnotatedDNAView.cpp:
```diff
@@ -865,7 +889,7 @@ QString AnnotatedDNAView::tryAddObject(GObject* o) {
 
             rCtx = findRelatedSequenceContexts(o);
-            if (rCtx.isEmpty()) {
+            if (rCtx.isEmpty() || d->relationIsSet) {
```
Shouldn't there be "!d->relationIsSet"?

atiunov replied:
Ok, it's move clear.

atiunov commented on src/plugins/GUITestBase/src/GTUtilsMdi.cpp:
```diff
@@ -264,7 +264,7 @@ void GTUtilsMdi::activateWindow(HI::GUITestOpStatus &os, const QString &windowNa
     GT_CHECK(mw != NULL, "MainWindow == NULL");
 
-    CHECK(!activeWindowTitle(os).contains(windowName), );
+    CHECK(!(activeWindowTitle(os) == windowName), );
```
Be ready to fix tests, that expect the previous behavior.

rasputinkirill replied:
In case of too many test fails, i'll change my scenario.

ichebyki committed:
Fix UGENE-6665 (#88)

* Fix UGENE-6665

* Fix UGENE-6665; Remove uppercase conversion; convert qstring to utf8; add check that any symbol in regexp is ASCII

ichebyki committed:
Fix UGENE-5371 (#86)

* Fix UGENE-5371

* Add new files to *.pro files; Add sam-files with non-ascii names support

* Add review notes

Commit:
Fix linux error for io includes

rasputinkirill committed:
UGENE-6760 PR issues fix

rasputinkirill committed:
UGENE-6760 PR issues fix

yalgaer committed:
Translation updates for WD settings dialog

rasputinkirill committed:
PR Issues fix

rasputinkirill committed:
PR Issues fix

rasputinkirill committed:
Merge branch 'UGENE-6760' of https://github.com/ugeneunipro/ugene into UGENE-6760

# Conflicts:
#	src/plugins/GUITestBase/src/GUITestBasePlugin.cpp
#	src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp
#	src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.h
